File: 786. K-th Smallest Prime Fraction.py
class Solution:
    def kthSmallestPrimeFraction(self, arr: List[int], k: int) -> List[int]:
        fractions = []
        for i in range(len(arr)):
            for j in range(i+1, len(arr)):
                fractions.append((arr[i], arr[j]))
        fractions.sort(key=cmp_to_key(compare))
        return fractions[k-1]

def compare(x,y):
    return x[0]*y[1] - x[1]*y[0]

File: 786. K-th Smallest Prime Fraction.java
class Solution {
    public int[] kthSmallestPrimeFraction(int[] arr, int k) {
        List<int[]> fractions = new ArrayList<>();
        for (int i = 0; i < arr.length; i++) {
            for (int j = i+1; j < arr.length; j++) {
                fractions.add(new int[]{arr[i], arr[j]});
            }
        }
        Collections.sort(fractions, (a,b) -> {
            int n1 = a[0], d1 = a[1], n2 = b[0], d2 = b[1];
            return n1*d2 - n2*d1;
        });
        return fractions.get(k-1);
    }
}
